sdk/src/main/scala/io/horizen/account/state/StateDbAccountStateView.scala

```diff
@@ -1,37 +1,34 @@
 package io.horizen.account.state
 
-import com.google.common.primitives.Bytes
 import io.horizen.SidechainTypes
 import io.horizen.account.proposition.AddressProposition
 import io.horizen.account.state.ForgerStakeMsgProcessor.AddNewStakeCmd
 import io.horizen.account.state.receipt.EthereumConsensusDataReceipt.ReceiptStatus
 import io.horizen.account.state.receipt.{EthereumConsensusDataLog, EthereumConsensusDataReceipt}
 import io.horizen.account.transaction.EthereumTransaction
-import io.horizen.account.utils.WellKnownAddresses.FORGER_STAKE_SMART_CONTRACT_ADDRESS
 import io.horizen.account.utils.{BigIntegerUtil, MainchainTxCrosschainOutputAddressUtil, WellKnownAddresses, ZenWeiConverter}
 import io.horizen.block.{MainchainBlockReferenceData, MainchainTxForwardTransferCrosschainOutput, MainchainTxSidechainCreationCrosschainOutput}
 import io.horizen.certificatesubmitter.keys.{CertifiersKeys, KeyRotationProof, KeyRotationProofTypes}
 import io.horizen.consensus.ForgingStakeInfo
+import io.horizen.evm.results.{EvmLog, ProofAccountResult}
+import io.horizen.evm.{Address, Hash, ResourceHandle, StateDB}
 import io.horizen.proposition.{PublicKey25519Proposition, VrfPublicKey}
 import io.horizen.transaction.mainchain.{ForwardTransfer, SidechainCreation}
 import io.horizen.utils.BytesUtils
-import io.horizen.evm.{Address, Hash, ResourceHandle, StateDB}
-import io.horizen.evm.results.{EvmLog, ProofAccountResult}
 import sparkz.crypto.hash.Keccak256
 import sparkz.util.SparkzLogging
 
 import java.math.BigInteger
-import java.util.Optional
 import scala.collection.JavaConverters.asScalaBufferConverter
 import scala.util.Try
 
 class StateDbAccountStateView(
     stateDb: StateDB,
-    messageProcessors: Seq[MessageProcessor]
+    messageProcessors: Seq[MessageProcessor],
+    var readOnly: Boolean = false
```

Comment:
It shouldn't happen, because any access to the state can be performed only by the NodeViewHolder actor, that is a single thread. However, `StateDbAccountStateView` is used for all the txs in a block, so a it is important that after a tx has been executed, the statedb should be set back to readwrite. This is guaranteed by a `finally` block at the end of `execute` method  of `StateTransition`.